tests/test_options.py

```diff
@@ -83,3 +86,28 @@ def test_list_of_urls(favicon_tags):
         == "https://secure.example.com/favicon/favicon-16x16.gif"
     )
     assert favicon_tags[0]["type"] == "image/gif"
+
+
+@pytest.mark.sphinx("html", testroot="static_files")
+def test_static_files(app, favicon_tags, favicon_tags_for_nested):
+
+    # this test should have 3 favicons
+    assert len(favicon_tags) == 2
+
+    # all favicons should have rel, href, type, and sizes attributes
+    for favicon_tag in chain(favicon_tags, favicon_tags_for_nested):
+        assert favicon_tag["rel"] == ["icon"]
+        assert "_static" in favicon_tag["href"]
+        assert favicon_tag["type"] == "image/svg+xml"
+        assert favicon_tag["sizes"]
+        assert "static-file" not in favicon_tag
+
+    for favicon_tag in favicon_tags:
+        assert favicon_tag["href"].startswith("_static")
+
+    for favicon_tag in favicon_tags_for_nested:
+        assert favicon_tag["href"].startswith("../_static")
+
+    static = Path(app.outdir, "_static")
+    assert (static / "square.svg").exists()
+    assert (static / "nested/triangle.svg").exists()
```

Comment:
Instead of having hardcoded test files, have you considered using the `tmp_path` fixture to generate all the test data in dedicated per-test dirs?

Reply:
Hi @webknjaz, I think there is some fixture-centric test setup going on, but it is using the [sphinx.testing.fixtures](https://www.sphinx-doc.org/en/master/internals/contributing.html?highlight=sphinx.testing.fixtures#unit-tests), which is not trivial from the point of view of regular pytest usage.

If we consider the `conftest` setup:
https://github.com/tcmetzger/sphinx-favicon/blob/a034cb8c5587d63bc599f0a33e91ff73f440e7b4/tests/conftest.py#L10-L12

`sphinx.testing.fixtures` is taking the directory under the `rootdir` and copying all the files to a temporary path, [(see `sphinx/testing/fixtures.py`)](https://github.com/sphinx-doc/sphinx/blob/121d9394fc5d33f13c66599620a9b9966c894b6a/sphinx/testing/fixtures.py#L102-L108).

Reply:
I am afraid avoiding the Sphinx helpers (that assume you create a directory skeleton in the disk instead of inside the file) and coding things by hand would make things more complicated.
Maybe we could have the best of both worlds, but I am not well versed enough in `sphinx.testing.fixtures` 😢. 

I really like the fact that the tests check for the existence of the files in a directory that is accessible by the page.

I can try to do the same check, without hardcoding the file names. But it will be a more convoluted test and more difficult to read/understand:

```python
    static = Path(app.outdir, "_static")
    original_dir = Path(__file__).parent / "rootdir/gfx"
    for image in original_dir.glob("**/*.svg"):
        relative = image.relative_to(original_dir)
        assert (static / relative).exists()
```